irohad/ametsuchi/tx_presence_cache_utils.hpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TX_PRESENCE_CACHE_UTILS_HPP
+#define IROHA_TX_PRESENCE_CACHE_UTILS_HPP
+
+#include "ametsuchi/tx_cache_response.hpp"
+#include "common/visitor.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+    /**
+     * Determine if transaction was already processed by its status
+     * @param tx_status - status obtained from transaction cache
+     * @return true if transaction was committed or rejected
+     */
+    inline bool isAlreadyProcessed(
+        const TxCacheStatusType &tx_status) noexcept {
+      return iroha::visit_in_place(
```

Comment:
This should call a constructor of `lambda_visitor` struct on each function call, which seems redundant. Maybe use boost::get instead?

Reply:
I think the constructor is going to be optimized away anyway, also boost::get throws and the logic would be less obvious